libs/utils/src/ballot_styles.ts

```diff
@@ -141,11 +130,36 @@ export function getDefaultLanguageBallotStyles(
  *
  *
  */
-export function getParentBallotStyles(
+
+export function getGroupedBallotStyles(
   ballotStyles: readonly BallotStyle[]
-): ParentBallotStyle[] {
-  return getDefaultLanguageBallotStyles(ballotStyles).map((bs) => ({
-    ...bs,
-    id: extractBallotStyleGroupId(bs.id),
-  }));
+): BallotStyleGroup[] {
+  const ballotStylesByGroupId = getBallotStyleGroupMap(ballotStyles);
+  return iter(ballotStylesByGroupId.keys())
+    .map((groupId) => {
+      const ballotStyleGroup = assertDefined(
+        ballotStylesByGroupId.get(groupId)
+      );
+      let englishBallotStyle: BallotStyle | undefined;
+      let legacyBallotStyle: BallotStyle | undefined;
+
+      for (const ballotStyle of ballotStyleGroup) {
+        if (deepEqual(ballotStyle.languages, [LanguageCode.ENGLISH])) {
+          englishBallotStyle = ballotStyle;
+        } else if (!ballotStyle.languages) {
+          legacyBallotStyle = ballotStyle;
+        }
+      }
+
+      const defaultBallotStyleInfo = assertDefined(
+        englishBallotStyle || legacyBallotStyle,
+        'Expected at least one English language ballot style per ballot style group.'
+      );
```

Comment:
All ballot styles in a group should only differ in the `languages` field - they should otherwise be identical. So it may just suffice in this function to pick a random ballot style from the group.